internal/pkg/aws/iam/iam.go

```diff
@@ -57,6 +62,18 @@ func (c *IAM) DeleteRole(roleARN string) error {
 	return nil
 }
 
+// CreateECSServiceLinkedRole creates a Service-Linked Role for Amazon ECS.
+// This role is necessary so that Amazon ECS can call AWS APIs.
+// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using-service-linked-roles.html
+func (c *IAM) CreateECSServiceLinkedRole() error {
+	if _, err := c.client.CreateServiceLinkedRole(&iam.CreateServiceLinkedRoleInput{
+		AWSServiceName: aws.String(ecsServiceName),
+	}); err != nil {
+		return fmt.Errorf("create service linked role for Amazon ECS: %w", err)
```

Comment:
Do we need to wrap this error, since it a) doesn't ever get reported and b) will always be wrapped by other methods?

Reply:
Hmm good question 🤔, I think:
a) That should be up to the client to decide whether they want to ignore the error or handle it, so we should still return the error.
b) It's definitely odd reading: `"create ECS service linked role: create ECS service linked role: AWS err"`
I think we should still wrap but I'll change the data of this error message so that it's more specific to the input of the call:
`"create service linked role for ecs.amazonaws.com: %w"`